particles/PipeApps2/source/CaptionThis.js

```diff
@@ -16,14 +16,30 @@ defineParticle(({DomParticle, html, log}) => {
     get template() {
       return `<span>{{json}}</span>`;
     }
-    render({query}) {
+    update({query}, state) {
       if (query) {
-        const text = query.name.split(' ');
-        const json = JSON.stringify(text);
-        this.updateSingleton('output', {json});
-        return {json};
+        this.classify(query.name);
       }
     }
-  };
 
+    async classify(text) {
+      this.startBusy();
+      const rawResponse =
```

Comment:
In general, we should use try/catch or try/finally to make sure 'doneBusy` is called when there is an exception or the system can get locked up.